independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InjectionPointsTransformer.java

```diff
@@ -35,12 +35,24 @@ public interface InjectionPointsTransformer extends BuildExtension {
     interface TransformationContext extends BuildExtension.BuildContext {
 
         /**
+         * This method is deprecated and marked for removal, use {@link #getAnnotationTarget()} instead.
+         * <p>
          * Returns {@link AnnotationTarget} representing this injection point.
+         * For injected parameters, this method returns the {@link AnnotationTarget} of the whole method.
          *
          * @return the annotation target of this injection point
          */
+        @Deprecated(forRemoval = true)
```

Comment:
I'm not sure if these are used outside of Quarkus. If so, please let's add a `since` version. I think you could use `3.12` as it looks highly likely this will hit `3.12`.

Reply:
+1 We should always add at least an estimate. I believe that we used the wording "at some point after..." or so.

Reply:
Good point

Reply:
Done